packages/xdl/src/TurtleApi.ts

```diff
@@ -0,0 +1,235 @@
+      },
+      extraOptions,
+      returnEntireResponse
+    );
+  }
+
+  async postAsync(
+    methodName: string,
+    data?: JSONObject,
+    extraOptions?: Partial<RequestOptions>,
+    returnEntireResponse: boolean = false
+  ) {
+    return this._requestAsync(
+      methodName,
+      {
+        httpMethod: 'post',
+        body: data,
+      },
+      extraOptions,
+      returnEntireResponse
+    );
+  }
+
+  async putAsync(
+    methodName: string,
+    data: JSONObject,
+    extraOptions?: Partial<RequestOptions>,
+    returnEntireResponse: boolean = false
+  ) {
+    return this._requestAsync(
+      methodName,
+      {
+        httpMethod: 'put',
+        body: data,
+      },
+      extraOptions,
+      returnEntireResponse
+    );
+  }
+
+  async deleteAsync(
+    methodName: string,
+    extraOptions?: Partial<RequestOptions>,
+    returnEntireResponse: boolean = false
+  ) {
+    return this._requestAsync(
+      methodName,
+      {
+        httpMethod: 'delete',
+      },
+      extraOptions,
+      returnEntireResponse
+    );
+  }
+
+  async _requestAsync(
+    methodName: string,
+    options: RequestOptions,
+    extraRequestOptions?: Partial<RequestOptions>,
+    returnEntireResponse: boolean = false
+  ) {
+    const url = `${_apiBaseUrl()}/${methodName}`;
+    let reqOptions: AxiosRequestConfig = {
+      url,
+      method: options.httpMethod,
+      headers: {
+        'fake-auth': JSON.stringify(
+          {
+            "userId": "85689f36-f570-459d-aca0-97e0e271283c",
+            "username": "turtle-v2-api-test-user"
+          }),
+      },
+    };
+
+    if (this.sessionSecret) {
+      reqOptions.headers['Expo-Session'] = this.sessionSecret;
+    }
+
+    // Handle qs
+    if (options.queryParameters) {
+      reqOptions.params = options.queryParameters;
+      reqOptions.paramsSerializer = QueryString.stringify;
+    }
+
+    // Handle body
+    if (options.body) {
+      reqOptions.data = options.body;
+    }
+
+    reqOptions = merge({}, reqOptions, extraRequestOptions);
+    let response;
+    let result;
+    try {
+      response = await axios.request(reqOptions);
+      result = response.data;
+    } catch (e) {
+      const maybeErrorData = idx(e, _ => _.response.data.errors.length);
+      if (maybeErrorData) {
+        result = e.response.data;
+      } else {
+        throw e;
+      }
+    }
+
+    if (result.errors && result.errors.length) {
+      let responseError = result.errors[0];
+      let error = new ApiV2Error(responseError.message, responseError.code);
+      error.serverStack = responseError.stack;
+      error.details = responseError.details;
+      throw error;
+    }
+
+    return returnEntireResponse ? response : result;
+  }
+
+  async uploadProject(bodyFormData: FormData) {
+    const url = `${_apiBaseUrl()}/upload`;
+    let reqOptions: AxiosRequestConfig = {
+      method: 'post',
+      url,
+      data: bodyFormData,
+      headers: {
+        'fake-auth': JSON.stringify(
```

Comment:
use correct auth token like in api and apiv2 and if you need to mock some behavior do this in server code of turtle-api